[KB] Google Workspace SAML setup (#1915)
semgrep/semgrep-docs: docs/kb/semgrep-appsec-platform/saml-google-workspace.md

---
description: Learn how to set up SAML access to Semgrep AppSec Platform with Google Workspace.
tags:
  - Semgrep AppSec Platform
  - SSO
---

# SAML Single Sign-on with Google Workspace

> The process of setting up SAML Single Sign-on for the Semgrep AppSec Platform with Google Workspace is fundamentally similar to the process for other providers. However, there are a few quirks of Google Workspace to account for.

Reviewer: Maybe this could just be something like, "This article demonstrates how to set up SAML Single Sign-on for Semgrep AppSec Platform with Google Workspace." I don't think we need to call out that there are quirks since we're pretty explicit in this article on how to set this up (I don't have strong opinions on this, though)

Author: That's fair. When I wrote the intro I think I hadn't decided if I was going to write a complete guide or not, but then I did. :)

Follow these steps:

> 1. [Set up a custom SAML app](https://support.google.com/a/answer/6087519?hl=en#zippy=%2Cstep-add-the-custom-saml-app) in Google Workspace. The default Name ID, primary email, is optimal for Semgrep AppSec Platform.

Reviewer: ```suggestion
1. [Set up a custom SAML app](https://support.google.com/a/answer/6087519?hl=en#zippy=%2Cstep-add-the-custom-saml-app) in Google Workspace. The default **Name ID** is the primary email, and this value is optimal for use with Semgrep AppSec Platform.
```

I'm not attached to this wording, but I think a bit more explanation would be helpful since I had to read this several times before understanding

> 2. When you reach step 12, **Add mapping**, add the two attribute statements that Semgrep AppSec Platform requires: `name` and `email`.

Reviewer: ```suggestion
2. When you reach the **Add mapping** step of the instructions to [Set up a custom SAML app](https://support.google.com/a/answer/6087519?hl=en#zippy=%2Cstep-add-the-custom-saml-app), add the two attribute statements that Semgrep AppSec Platform requires: `name` and `email`.
```

Rewriting this a bit in case the step number ever changes

Author: I like that part, but the extra link seems weird to me since we just linked in the previous line. I would assume the person has the Google steps open here already. What do you think about that?

Reviewer: That's fair; I updated the suggestion to omit the link

   * The attribute mapped to `email` should be the primary email.
   * The attribute mapped to `name` should be some form of the user's name. You can use a default attribute like the user's first name, or create a custom attribute for their full name.
      ![Attribute mappings](/img/kb/google_attributes.png)
3. Sign in to Semgrep AppSec Platform.
4. Navigate to **[Settings > Access > Login methods](https://semgrep.dev/orgs/-/settings/access/loginMethods)**.
5. Click **Add SSO configuration** and select **SAML2 SSO**.

> 6. Provide a **Display name** and your email domain.

Reviewer: ```suggestion
6. Provide a **Display name** and your **Email domain**.
```

> 7. Copy the **SSO URL** and **Audience URL (SP Entity ID)**, and provide them to Google Workspace as the **ACS URL** and **Entity ID** respectively.

Reviewer: ```suggestion
7. Copy the **SSO URL** and **Audience URL (SP Entity ID)**, and provide them to Google Workspace as the **ACS URL** and **Entity ID**, respectively.
```

> 8. If you have not already, from the custom SAML app in Google Workspace, copy your IDP metadata, including the SSO URL and Entity ID and the x509 certificate.

Author: I like the extra second phrase, but the reason I wrote the other bit in is that Google's instructions have you do it much earlier. Is there a better way to handle that, or do you think it's not worth worrying about?

> 8. If you have not already, from the custom SAML app in Google Workspace, copy your IDP metadata, including the SSO URL and Entity ID and the x509 certificate.

Reviewer: I don't think it's worth worrying about -- my assumption here is that if they've already copied the info, they'll just proceed to the second step of the instruction

> 9. Enter these in Semgrep AppSec Platform as the **IdP SSO URL** and **IdP Issuer ID** values respectively, and upload or paste the X509 Certificate.

Reviewer: Maybe consolidate this step with the previous step?

Author: To me these are separate things, you first get the data, then you add it.
